data/src/main/java/com/microsoft/azure/kusto/data/format/CslDateTimeFormat.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.azure.kusto.data.format;
+
+import com.microsoft.azure.kusto.data.Ensure;
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Date;
+
+public class CslDateTimeFormat extends CslFormat {
+    public static final String KUSTO_DATETIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSSSSS'Z'";

```

Comment:
IIRC kusto can process partial datetime strings, e.g. this is valid query
`KustoLogs
| where Timestamp  between (datetime("2021") .. 20m)`

Reply:
You're right, the Csl parser in C# is much more mature and can handle things like that. I'd suggest that achieving full feature parity with C# is out of scope in this PR, but I tried to reasonably mimic the C# DateTime behavior.